### ISSUE TITLE
Pass invalid values to the onDateChange prop

### ISSUE THREAD

**Issue by marcgarygray** [airbnb/react-dates#1937]
I believe it would be helpful to a lot of developers to be able to know if an invalid date has been typed by the user into the `<input type="text">` node. In my opinion, the most logical place to provide that information would be in the `onDateChange` handler. I could see an argument for a simple `boolean` that's `false` if the input is not a valid `moment.Moment` or passing the invalid input as a `string`.
I know that an invalid input just passes `null` as the date to the handler, but that being the case, there's no way to differentiate between the user simply clearing the data (a valid interaction) versus entering text that can't be parsed into a valid date (an invalid interaction).

If you allow pull requests from the masses, I would be more than happy to make the change and submit a PR.

**Comment by marcgarygray** [comment]
### Note
This is for the `<SingleDatePicker>` specifically. It's the only component from this library that I've utilized thus far. I'm not sure if this is the same or different across all of the components exported by the library.

**Comment by ljharb** [comment]
If there's a nonbreaking way to also pass the original input string, I'd be open to that.

I think this might be a duplicate of #1023?

**Comment by marcgarygray** [comment]
@ljharb 
Are you open to outside contributions? I'm happy to make the change.
If so, any guidelines (PR must come with a test, etc., etc.) are appreciated. :)

**Comment by ljharb** [comment]
Yes, absolutely! Tests are a requirement; a storybook story is nice but not required.

It'd be ideal to make this change on every component in the library, so that everything can follow similar patterns.

**Comment by marcgarygray** [comment]
@ljharb working on this tonight.
Looks like your workflow is to create a branch named something like username/issue#/description and make a PR to merge that branch to `master`, yeah?

**Comment by ljharb** [comment]
The workflow in general terms for making a PR to any open source project is to fork the repo, make a branch on the fork with your changes, and then make a PR to merge that to the source's default branch (usually `master`).

**Comment by marcgarygray** [comment]
@ljharb thanks. Submitted. 👍
